components/LoadingContainer.js

```diff
@@ -0,0 +1,76 @@
+import React, { useMemo, useState } from 'react';
+import LottieView from 'lottie-react-native';
+import PropTypes from 'prop-types';
+import { connectStyle } from '@shoutem/theme';
+import { View } from '@shoutem/ui';
+import { animations } from '../assets';
+
+function LoadingContainer({
+  animation,
+  children,
+  loading,
+  animationScale,
+  style,
+}) {
+  const [dimensions, setDimensions] = useState({});
+
+  function onLayoutChange(event) {
+    const { width, height } = event.nativeEvent.layout;
```

Comment:
Is this worth using `useCallback` on with `[event.nativeEvent.layout.width, event.nativeEvent.layout.height]`?

Trying to learn, more a question than a suggestion.

Reply:
event is received through function props and it doesn't belong to dependency array - it can't be there - breaks the app with "Can't find variable: event".
Only constants present in component can be depended on